src/cocoa/toga_cocoa/widgets/textinput.py

```diff
@@ -65,3 +70,11 @@ def rehint(self):
 
     def set_on_change(self, handler):
         pass
+
+    def set_error(self, error_message):
+        self.interface.value = self.last_valid_value
+        if self.interface.window is not None:
+            self.interface.window.error_dialog("Validation Error", error_message)
+
+    def unset_error(self):
```

Comment:
Unset error *should* be a no-op on Cocoa, as the input can never be in a truly "error" state - correct?